CaptiveRequestHandler.h

```diff
@@ -0,0 +1,184 @@
+            String json = "";
+            for (int i = 0; i < len; i++)  json += char(data[i]);
+  
+            StaticJsonDocument<1024> credentialsJsonDoc;
+            if (!deserializeJson(credentialsJsonDoc, json)) {
+              if(setCredentials(credentialsJsonDoc.as<JsonObject>())) request->send(200);
+
+              else request->send(400);
+            }
+          }
+          else if(request->url() == "/reboot") {
+              String json = "";
+              for (int i = 0; i < len; i++)  json += char(data[i]);
+    
+              StaticJsonDocument<256> rebootJsonDoc;
+              if (!deserializeJson(rebootJsonDoc, json)) {
+                //TODO:
+                // int delayMs = rebootJsonDoc["delay"];
+
+                // softReset(delayMs);
+                // socket_server.textAll("RESTART");
+              }
+
+              request->send(200);
+          }
+          else {
+            request->send(404);
+          }
+        }
+      //}
+    }
+
+    void sendFile(AsyncWebServerRequest * request, String path) {
+      Serial.println("handleFileRead: " + path);
+
+      if (SPIFFS.exists(path)) {
+        request->send(SPIFFS, path, getContentType(path));
+      }
+      else {
+        request->send(404);
+      }
+    }
+
+    String getContentType(String filename) {
+      if (filename.endsWith(".htm")) return "text/html";
+      else if (filename.endsWith(".html")) return "text/html";
+      else if (filename.endsWith(".css")) return "text/css";
+      else if (filename.endsWith(".js")) return "application/javascript";
+      else if (filename.endsWith(".png")) return "image/png";
+      else if (filename.endsWith(".gif")) return "image/gif";
+      else if (filename.endsWith(".jpg")) return "image/jpeg";
+      else if (filename.endsWith(".ico")) return "image/x-icon";
+      else if (filename.endsWith(".xml")) return "text/xml";
+      else if (filename.endsWith(".pdf")) return "application/x-pdf";
+      else if (filename.endsWith(".zip")) return "application/x-zip";
+      else if (filename.endsWith(".gz")) return "application/x-gzip";
+      else if (filename.endsWith(".json")) return "application/json";
+      return "text/plain";
+    }
+
+    void getCredentials(AsyncWebServerRequest *request) {
+      Serial.println("getCredentials");
+      AsyncResponseStream *response = request->beginResponseStream("application/json");
+
+      StaticJsonDocument<1024> settingsJsonDoc;
+      //TODO:
+      /*
+      settingsJsonDoc["local_mac"] = myID;
+      settingsJsonDoc["local_ssid"] = "";
+      settingsJsonDoc["local_pass_len"] = 0; //local_pass.length;
+      settingsJsonDoc["remote_ssid"] = "";
+      settingsJsonDoc["remote_pass_len"] = 0; //remote_pass.length;
+      settingsJsonDoc["remote_mac"] = getRemoteMacAddress(1);
+      settingsJsonDoc["local_paired_status"] = getCurrentPairedStatusAsString();
+      Serial.println(getCurrentPairedStatusAsString());
+      */
```

Comment:
Is it worth breaking out the get / set functions for preferences into a separate class so that it can be called from here, or could the CaptiveRequestHandler just have access to YoYoNetworkManager and call like `YoYoNetworkManager::getRemoteMacAddress(1)` ?